### ISSUE TITLE
Fix component show/hide is incorrectly disabled (e.g. on `GeometryPath`)

### ISSUE THREAD

**Issue by tgeijten** [ComputationalBiomechanicsLab/opensim-creator#958]
In version 0.5.16, if I right-click on a muscle and go to Display, the "show" and "hide" etc. options are grayed out / disabled.

**Comment by adamkewley** [comment]
Unfortunately, this is dictated by whether the `Component`, or any sub-`Component` has an `Appearance` property in its OpenSim definition. It needs to be this way (at the moment, at least) so that when the model is saved/undone/redone the same set of visible components is reconstructed as-is (as opposed to OpenSim GUI, which will inject additional visualization etc. at the UI level, but those aren't encode-able, so users need to be trained to know "which ones count").

The only way to fix this (imo) is to change how visibility is handled at the OpenSim level. It should probably be something that can be tagged onto any component and applied hierarchically to the tree. Changing that would require also updating OpenSim GUI to be aware of the possible association (e.g. it'd be added to a base class somewhere and then OpenSim GUI or the decoration-generation code would need to know to look for it).

I agree, though, it's bad UX. A lot of the visibility-toggling stuff in OpenSim is (e.g. frame toggling is at the model level and completely distinct from `Appearance`), but I *personally* don't think it's a good idea for OSC to use custom rules that can't be encoded into the model itself, and dragging around non-encoded UI state is usually a recipe for confusion ("dear Adam, I hidden X but it still shows in OpenSim GUI").

I'll add a comment explaining the situation then that button is greyed out - best I can do until the underlying problem is fixed.

**Comment by tgeijten** [comment]
That's weird -- is that a bug or regression in OpenSim then? Both muscles and bodies are affected by this, and both of those have Appearance properties (as they obviously should).

**Comment by adamkewley** [comment]
Ah, you're right - it's a regression. I thought you were referrring to right-clicking the path points rather than the `GeometryPath`.

I'll investigate the regression and get it sorted!

**Comment by adamkewley** [comment]
This has now been fixed. It likely happened during a recent refactor in which I made model editing/readonly a runtime thing (so that UI components can be reused between readonly stuff like simulator tabs and the editor tab). The logic (fixed in f70ff71) was inverted, because I probably though the UI layer takes a `isDisabled` rather than `isDisabled` flag variable.

Thanks for reporting it @tgeijten
